internal/resolver/config_selector.go

```diff
@@ -119,7 +119,10 @@ type ClientInterceptor interface {
 
 // ServerInterceptor is unimplementable; do not use.
 type ServerInterceptor interface {
-	notDefined()
+	// AllowedToProceed checks if an incoming RPC is allowed to proceed based on
+	// information about connection RPC was received on, and HTTP Headers. This
+	// information will be piped into context.
+	AllowRPC(ctx context.Context) error // TODO: Make this a real interceptor for filters such as rate limiting.
```

Comment:
What does this TODO involve? If rate limiting is coming soon, does it make sense to actually make it a real interceptor right now?

Reply:
No. Me and Doug decided to make this API Definition in the concept of as little functionality as needed. Thus, he told me to add a TODO for when we need to add a different type of filter, since this API is specific to RBAC, a logical gatekeeper at the beginning of the RPC.